src/l500/l500-options.cpp

```diff
@@ -111,15 +163,45 @@ namespace librealsense
 
             auto resolution_option = std::make_shared<sensor_mode_option>(this, option_range{ RS2_SENSOR_MODE_VGA,RS2_SENSOR_MODE_COUNT - 1,1, default_sensor_mode }, "Notify the sensor about the intended streaming mode. Required for preset ");
 
+            // changing the resolution affects the defaults
+            resolution_option->add_observer( [&]( float ) {
+
+                for( auto opt : _hw_options )
+                    opt.second->update_default();
+
+                auto p = calc_preset_from_controls();
+                if( p != RS2_L500_VISUAL_PRESET_CUSTOM )
+                    _preset->set_with_no_signal( p );
+
+                set_controls_defaults();
+            } );
+
+
             depth_sensor.register_option(RS2_OPTION_SENSOR_MODE, resolution_option);
 
+             _digital_gain = register_option< uvc_xu_option< int >,
```

Comment:
Will this also change the location of the option in the Viewer?

Reply:
no the order is by drawing_order for some options and for the others is by the get_supported_options() order.